### ISSUE TITLE
In NodeJS, Blob trigger always binds as binary buffer regardless of 'dataType' setting

### ISSUE THREAD

**Issue by adamstoffel** [Azure/azure-functions-nodejs-worker#517]
The Blob Trigger binding for NodeJS functions always binds data as a Buffer even if 'dataType' is set to 'stream' or 'string'.

#### Repro steps

1. Create a node-based function using the blob trigger
2. Set "dataType" to "stream" in function.json:
```json
{
  "bindings": [
    {
      "name": "fileBlob",
      "type": "blobTrigger",
      "direction": "in",
      "path": "container/{fileName}.csv",
      "connection": "AzureWebJobsStorage",
      "dataType": "stream"
    }
  ],
  "scriptFile": "../dist/myFunction/index.js"
}

```

#### Expected behavior

`fileBlob` input parameter should be some kind of Stream type

#### Actual behavior

`fileBlob` input parameter is a Buffer

#### Known workarounds

None

#### Related information 

Functions runtime ~3
Node ~14

**Comment by ejizba** [comment]
The Node.js worker doesn't support streaming data. Here is an issue on the host repo with some of the background: https://github.com/Azure/azure-functions-host/issues/1361

We might be able to make progress on this, though. @gohar94 implemented a 'shared memory' feature that is meant to help with memory usage in general, although I don't know if it necessarily passes a stream to the users's code. It's only implemented for the Python worker so far. @gohar94 does the Python implementation pass along a stream to the user? Could or should we do that when implementing this feature for Node.js?

More info:
- Shared memory issue: https://github.com/Azure/azure-functions-host/issues/6791
- Shared memory PR: https://github.com/Azure/azure-functions-host/pull/6836
- Python docs: https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-python#shared-memory

**Comment by gohar94** [comment]
Hi @ejizba - the shared memory feature on Python passes the whole set of bytes to the user function as of today. However, I do think that some stream-like abstraction can be given to the user (like `io.BytesIO` over the `mmap`). It will likely still all be loaded in memory but the user could operate on those bytes as a stream, if that's what you're asking. 
Happy to take a look if you decide to do this for Node.js.

**Comment by ejizba** [comment]
Yeah I'm less interested in doing a stream if it's just an abstraction, not an actual stream with all the benefits.

How likely is the shared memory feature to help with 'out of memory' issues? Or is it realistically just for performance?

**Comment by gohar94** [comment]
> Yeah I'm less interested in doing a stream if it's just an abstraction, not an actual stream with all the benefits.
> 
> How likely is the shared memory feature to help with 'out of memory' issues? Or is it realistically just for performance?

I believe it does reduce one extra copy (which gRPC creates when transferring between the host and the worker).
It does seem possible to do full streaming from the host to the worker using shared memory but it would involve some more work (like communicating what pieces of data become available and which ones have been consumed etc.) Would be an interesting project though.

**Comment by MarioArriaga92** [comment]
Is there an ETA on this?

**Comment by ejizba** [comment]
@MarioArriaga92 no there is not an ETA for this. We want to do it, but it will not be simple and so far other work has been higher priority. Specifically, you can see our roadmap here with the work we currently have prioritized:
https://github.com/Azure/azure-functions-nodejs-worker/wiki/Roadmap

**Comment by sig9** [comment]
Having to consider other hosting options due to this limitation.
My use case is to implement a server api in node that calls an OpenAI api function which returns a Server-Sent Events stream. I would like to relay those responses to the browser. I have a minimal git repo that demonstrates the issue. No deployment needed to demonstrate this, the azure-functions-core-tools package behaves the same way as a deployed app.
https://github.com/sig9/nuxt-app-sse/
